ygen/genir.go

```diff
@@ -22,6 +32,32 @@ type IROptions struct {
 	// Transformation options specifies any transformations that should
 	// be applied to the input YANG schema when producing the IR.
 	TransformationOptions TransformationOpts
+
+	// NestedDirectories specifies whether the generated directories should
+	// be nested in the IR.
+	NestedDirectories bool
+
+	// AbsoluteMapPaths specifies whether the path annotation provided for
+	// each field should be relative paths or absolute paths.
+	AbsoluteMapPaths bool
+
+	// AppendEnumSuffixForSimpleUnionEnums appends an "Enum" suffix to the
```

Comment:
Is it worth cutting our v1 at the point that we have integrated the IR? I'm just wondering whether we're carrying complexity that we could potentially avoid here?

Reply:
Yes we are carrying complexity, but it's not a whole lot. My thinking on v1 right now is to get all the pending v1 work done so that we can drop as much complexity at that time and it'll also mean we'll likely maintain it for a while. Otherwise it either means we'll still have to carry the complexity from having to maintain those flags, or if we release v2 too soon I think it wouldn't give people a good sense of security about the codebase.

Reply:
The cost here isn't particularly high as you say -- so I'm happy with just saying that this can stick around for some time.